clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -10794,6 +10794,23 @@ def err_non_local_variable_decl_in_for : Error<
   "declaration of non-local variable in 'for' loop">;
 def err_non_variable_decl_in_for : Error<
   "non-variable declaration in 'for' loop">;
+
+def ext_c23_non_local_variable_decl_in_for : Extension<
```

Comment:
I wonder if we're missing some feature marked as supported on https://clang.llvm.org/c_status.html ? @AaronBallman

Reply:
This change happened sometime between N3047 and N3096, so it was a national body comment. It turns out it was GB-125 (from https://www.open-std.org/jtc1/sc22/wg14/www/docs/n3067.doc), which was addressed by section 4.3 in https://www.open-std.org/jtc1/sc22/wg14/www/docs/n3078.htm according to https://www.open-std.org/jtc1/sc22/wg14/www/docs/n3091.doc

So no need to track it on the status page, it wasn't really a feature paper but a cleanup.

Reply:
You should remove the error diagnostics because they're no longer used.

Reply:
> You should remove the error diagnostics because they're no longer used.

Well, they *are* still used for C++ range-based for loops and for whatever an `ObjCForCollectionStmt` is.

Reply:
Oh good call, I missed that!